Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/AssetBrowserExpandedTableViewProxyModel.cpp

```diff
@@ -0,0 +1,197 @@
+        {
+            if (orientation == Qt::Horizontal && role == Qt::DisplayRole)
+            {
+                section += section ? static_cast<int>(AssetBrowserEntry::Column::Type) - 1 : 0;
+                return tr(AssetBrowserEntry::m_columnNames[section]);
+            }
+
+            return QVariant();
+        }
+
+        bool AssetBrowserExpandedTableViewProxyModel::hasChildren(const QModelIndex& parent) const
+        {
+            if (parent != m_rootIndex)
+            {
+                return false;
+            }
+            return true;
+        }
+
+        int AssetBrowserExpandedTableViewProxyModel::columnCount([[maybe_unused]]const QModelIndex& parent) const
+        {
+            return 6;
+        }
+
+        void AssetBrowserExpandedTableViewProxyModel::SetRootIndex(const QModelIndex& index)
+        {
+            m_rootIndex = index;
+        }
+
+        bool AssetBrowserExpandedTableViewProxyModel::GetShowSearchResultsMode() const
+        {
+            return m_searchResultsMode;
+        }
+
+        void AssetBrowserExpandedTableViewProxyModel::SetShowSearchResultsMode(bool searchMode)
+        {
+            if (m_searchResultsMode != searchMode)
+            {
+                m_searchResultsMode = searchMode;
+                beginResetModel();
+                endResetModel();
+            }
+        }
+        inline constexpr auto operator"" _hash(const char* str, size_t len)
+        {
+            return AZStd::hash<AZStd::string_view>{}(AZStd::string_view{ str, len });
+        }
+
+        const AZStd::string AssetBrowserExpandedTableViewProxyModel::ExtensionToType(AZStd::string_view str) const
+        {
+            switch (AZStd::hash<AZStd::string_view>{}(str))
+            {
+            case ".png"_hash:
+                return "PNG";
+            case ".scriptcanvas"_hash:
+                return "Script Canvas";
+            case ".fbx"_hash:
+                return "FBX";
+            case ".mtl"_hash:
+                return "Material";
+            case ".animgraph"_hash:
+                return "Anim Graph";
+            case ".motionset"_hash:
+                return "Motion Set";
+            case ".assetinfo"_hash:
+                return "Asset Info";
+            case ".py"_hash:
+                return "Python Script";
+            case ".lua"_hash:
+                return "Lua Script";
+            case ".tif"_hash:
+                return "TIF";
+            case ".physxmaterial"_hash:
+                return "PhysX Material";
+            case ".prefab"_hash:
+                return "Prefab";
+            case ".dds"_hash:
+                return "DDS";
+            case ".font"_hash:
+                return "Font";
+            case ".xml"_hash:
+                return "XML";
+            case ".json"_hash:
+                return "JSON";
+            case ".exr"_hash:
+                return "EXR";
+            case ".wav"_hash:
+                return ".WAV";
```

Comment:
Should `WAV` have the `.` removed as well?

Reply:
Yes. Done.